### PR TITLE
refactor: renaming tracking service.MLflowService to client.MLflowClient

### PR DESCRIPTION
**Pull request by mparkhe**
- `mlflow.tracking.service.MLflowService` to `client.MLflowClient`
- removing `get_service` and having a Java style constructor to instantiate `MLflowClient`

### CODE REVIEW

File: mlflow/tracking/client.py [hunk 1]
```diff
@@ -17,12 +17,13 @@
 _DEFAULT_USER_ID = "unknown"
 
 
-class MLflowService(object):
+class MLflowClient(object):
```
Comment (andrewmchen) [inline mlflow/tracking/client.py]: `MlflowClient` is consistent with the Java API

File: mlflow/tracking/client.py [hunk 2]
```diff
@@ -17,12 +17,13 @@
...
     """Client of an MLflow Tracking Server that creates and manages experiments and runs.
```
Comment (andrewmchen) [inline mlflow/tracking/client.py]: Can we move the `get_service` docstring here?

**Review by tomasatdatabricks — APPROVED** [review]
Looks good, thanks Mani. 

I only have a (nit) comment: Personally, I prefer calling a function to get a new client object instead of calling constructor directly so that the constructor does not depend on the global state (which I would not expect).

File: mlflow/tracking/client.py [hunk 3]
```diff
@@ -197,19 +198,6 @@ def set_terminated(self, run_id, status=None, end_time=None):
                                    end_time=end_time)
 
 
-def get_service(tracking_uri=None):
```
Comment (andrewmchen) [inline mlflow/tracking/client.py]: There's still a reference to `get_service` at `docs/source/tracking.rst`.

**Review by andrewmchen — CHANGES_REQUESTED** [review]

File: mlflow/tracking/client.py (hunk 2 above)
Comment (mparkhe) [reply mlflow/tracking/client.py]: :+1:

File: mlflow/tracking/client.py (hunk 3 above)
Comment (mparkhe) [reply mlflow/tracking/client.py]: Good catch! Thanks...

**Comment by mparkhe** [comment]
@tomasatdatabricks : this does not change the behavior, i.e. if `tracking_uri` is not provided it is picked up from env variables like before. If needed we can change this functionality in future. The objective is to match how pythonic and java client constructor call.

**Review by aarondav — APPROVED** [review]

**Review by andrewmchen — APPROVED** [review]

File: tests/projects/test_databricks.py [hunk 4]
```diff
@@ -93,8 +93,8 @@ def before_run_validations_mock():  # pylint: disable=unused-argument
 
 @pytest.fixture()
 def set_tag_mock():
-    with mock.patch("mlflow.projects.databricks.tracking.get_service") as m:
-        mlflow_service_mock = mock.Mock(wraps=get_service())
+    with mock.patch("mlflow.projects.databricks.tracking.MLflowClient") as m:
```
[the remark this answers is not in the document]
Comment (mparkhe) [reply tests/projects/test_databricks.py]: Thanks

**Pull request merged**